src/Common/Crypto/CryptoGen.php

```diff
@@ -449,84 +454,139 @@ public function aes256Decrypt_mycrypt(string $sValue): string
+    }
 
-        // Collect key
-        if ($keySource == 'database') {
-            $sqlKey = $this->sqlQueryNoLog("SELECT `value` FROM `keys` WHERE `name` = ?", [$label]);
-            $key = base64_decode($sqlKey['value']);
-        } else { //$keySource == 'drive'
-            if (($version == "one") || ($version == "two") || ($version == "three") || ($version == "four")) {
-                // older key versions that did not encrypt the key on the drive
-                $key = base64_decode(rtrim($this->fileGetContents($GLOBALS['OE_SITE_DIR'] . "/documents/logs_and_misc/methods/" . $label)));
-            } else {
-                $key = $this->decryptStandard($this->fileGetContents($GLOBALS['OE_SITE_DIR'] . "/documents/logs_and_misc/methods/" . $label), null, 'database');
-            }
+    /**
+     * Create a new crypto key in the database.
+     * Produce a 256bit key (32 bytes equals 256 bits)
+     *
+     * @param  string $label   the version and sub-version label
+     * @param  string $version the key encryption version (not used in this implementation, but important for future compatibility)
+     * @return string the plaintext key value
+     * @throws CryptoGenException if key creation, storage, encoding or encryption fails
+     */
+    protected function createDatabaseKey(string $label, string $version): string
+    {
+        // Produce a 256bit key (32 bytes equals 256 bits)
+        $newKey = $this->getRandomBytes(32);
+        if (empty($newKey)) {
+            throw new CryptoGenException("OpenEMR Error: Random Bytes error - exiting");
+        }
+        $this->sqlStatementNoLog("INSERT INTO `keys` (`name`, `value`) VALUES (?, ?)", [$label, base64_encode($newKey)]);
+        // round trip to be sure the newly created key is correctly stored and encoded
+        // this shouldn't cause a performance problem because new keys aren't created too frequently.
+        $sqlResult = $this->sqlQueryNoLog("SELECT `value` FROM `keys` WHERE `name` = ?", [$label]);
+        $sqlKey = base64_decode((string) $sqlResult['value']);
+        if ($newKey === $sqlKey) {
+            return $newKey;
+        }
+        throw new CryptoGenException("OpenEMR Error: The newly created key could not be stored or encoded correctly.");
+    }
+
+    /**
+     * Collect the crypto key from the drive, creating it if necessary.
+     *
+     * @param  string $label   the version and sub-version label
+     * @param  string $version the key encryption version
+     * @return string the plaintext key value
+     * @throws CryptoGenException if key collection fails
+     */
+    protected function collectDriveKey(string $label, string $version): string
+    {
+        $keyPath = $GLOBALS['OE_SITE_DIR'] . "/documents/logs_and_misc/methods/" . $label;
+        $isLegacyVersion = in_array($version, ["one", "two", "three", "four"]);
+        if (!$this->fileExists($keyPath)) {
+            $key = $this->createDriveKey($label, $version);
```

Comment:
looks like the $key variable here doesn't do anything (either skipped when key already exists or is overwritten below)